unstructured/partition/ocr.py

```diff
@@ -383,84 +437,66 @@ def zoom_image(image: PILImage, zoom: float = 1) -> PILImage:
     return PILImage.fromarray(new_image)
 
 
-def get_ocr_layout_from_image(
+def get_ocr_data_from_image(
     image: PILImage,
     ocr_languages: str = "eng",
-    ocr_agent: str = "tesseract",
-) -> List[TextRegion]:
-    """
-    Get the OCR layout from image as a list of text regions with paddle or tesseract.
-    """
-    if ocr_agent == "paddle":
-        logger.info("Processing OCR with paddle...")
+    ocr_agent: str = OCR_AGENT_TESSERACT,
+    output_type: OCROutputType = OCROutputType.STRING,
+) -> Union[str, List[TextRegion]]:
+    if ocr_agent == OCR_AGENT_PADDLE:
+        logger.info("Processing entrie page OCR with paddle...")
         from unstructured.partition.utils.ocr_models import paddle_ocr
 
         # TODO(yuming): pass in language parameter once we
         # have the mapping for paddle lang code
+        # see CORE-2034
         ocr_data = paddle_ocr.load_agent().ocr(np.array(image), cls=True)
         ocr_layout = parse_ocr_data_paddle(ocr_data)
+        if output_type == OCROutputType.STRING:
+            output_data = "\n\n".join([r.text for r in ocr_layout])
+        else:
+            output_data = ocr_layout
     else:
-        logger.info("Processing OCR with tesseract...")
-        zoom = 1
-        ocr_df: pd.DataFrame = unstructured_pytesseract.image_to_data(
-            np.array(image),
-            lang=ocr_languages,
-            output_type=Output.DATAFRAME,
-        )
-        ocr_df = ocr_df.dropna()
-
-        # tesseract performance degrades when the text height is out of the preferred zone so we
-        # zoom the image (in or out depending on estimated text height) for optimum OCR results
-        # but this needs to be evaluated based on actual use case as the optimum scaling also
-        # depend on type of characters (font, language, etc); be careful about this
-        # functionality
-        text_height = ocr_df[TESSERACT_TEXT_HEIGHT].quantile(
-            env_config.TESSERACT_TEXT_HEIGHT_QUANTILE,
-        )
-        if (
-            text_height < env_config.TESSERACT_MIN_TEXT_HEIGHT
-            or text_height > env_config.TESSERACT_MAX_TEXT_HEIGHT
-        ):
-            # rounding avoids unnecessary precision and potential numerical issues assocaited
-            # with numbers very close to 1 inside cv2 image processing
-            zoom = np.round(env_config.TESSERACT_OPTIMUM_TEXT_HEIGHT / text_height, 1)
-            ocr_df = unstructured_pytesseract.image_to_data(
-                np.array(zoom_image(image, zoom)),
+        if output_type == OCROutputType.STRING:
+            output_data = unstructured_pytesseract.image_to_string(
+                np.array(image),
```

Comment:
we might also want to apply the same scaling logic to `image_to_string` as `image_to_data` below, but this could be an additional PR

Reply:
Seems like a good catch! We'll consider it in an additional PR. Thanks!